dashboard/lib/services/partial_registration/user_builder.rb

```diff
@@ -17,15 +17,32 @@ def call
 
       private def user_params
         user_params = ActionController::Parameters.new(params).require(:user).permit(new_user_permitted_params)
+        user_params[:locale] = request.locale
         user_params[:user_type] = user_params[:user_type].presence || session[:default_sign_up_user_type]
         case user_params[:user_type]
         when ::User::TYPE_TEACHER
           user_params[:age] = '21+'
           user_params[:email_preference_opt_in_required] = true
-          user_params[:email_preference_opt_in] = user_params[:email_preference_opt_in] == 'true' ? 'yes' : 'no'
+          user_params[:email_preference_opt_in] = ActiveModel::Type::Boolean.new.cast(user_params[:email_preference_opt_in]) ? 'yes' : 'no'
           user_params[:email_preference_request_ip] = request.ip
           user_params[:email_preference_source] = EmailPreference::ACCOUNT_SIGN_UP
           user_params[:email_preference_form_kind] = '0'
+
+          if user_params[:school_info_attributes].present?
+            if SharedConstants::NON_SCHOOL_OPTIONS.to_h.value?(user_params[:school_info_attributes]['schoolId'])
+              user_params[:school_info_attributes]['schoolId'] = nil
+            end
+            school_params = {
```

Comment:
Could most of this be done using `transform_keys(&:underscore)`?

So, I'm thinking lines 32-24 stay the same then the rest of this is something like
```
user_params[:school_info_attributes].transform_keys!(&:underscore).permit(:school_id, :school_name, :school_type, :school_zip, :school_state, :country, :full_address)
```
Might need to play with the syntax a bit, though.